inc/Provider.php

```diff
@@ -130,6 +131,16 @@ final public function request_items( array $args ) : MediaList {
 			return $items;
 		}
 
+		if ( isset( $args['posts_per_page'] ) && ( $args['posts_per_page'] > 0 ) && count( $array ) > $args['posts_per_page'] ) {
+			throw new Exception(
```

Comment:
How about using `RangeException` instead? (This would require a namespace import at the top of the file.)